src/tracing/mod.rs

```diff
@@ -280,18 +300,6 @@ impl TracingInspector {
             PushTraceKind::PushAndAttachToParent
         };
 
-        if self.trace_stack.is_empty() {
-            // this is the root call which should get the original gas limit of the transaction,
-            // because initialization costs are already subtracted from gas_limit
-            // For the root call this value should use the transaction's gas limit
-            // See <https://github.com/paradigmxyz/reth/issues/3678> and <https://github.com/ethereum/go-ethereum/pull/27029>
-            gas_limit = context.env.tx.gas_limit;
-
-            // we set the spec id here because we only need to do this once and this condition is
-            // hit exactly once
-            self.spec_id = Some(context.spec_id());
```

Comment:
this got lost, or do we no longer need this?

Reply:
Unused, we also have unused variables in the various constructors

Reply:
I double check the code. This is `self.spec_id` for `TracingInspector` is used to set the gas limit for the Parity trace root only once. For Geth Debug trace, this will be set in the newly constructed function `set_transaction_gas_limit()` and `with_transaction_gas_limit()`. Therefore, we on longer need this.